Sming/Arch/Esp32/Tools/install.sh

```diff
@@ -31,18 +31,20 @@ $PKG_INSTALL ${PACKAGES[*]}
 if [ -d "$IDF_PATH" ]; then
     printf "\n\n** Skipping ESP-IDF clone: '$IDF_PATH' exists\n\n"
 else
-    git clone -b release/v4.1 https://github.com/espressif/esp-idf.git $IDF_PATH
+    git clone -b release/v4.3 https://github.com/espressif/esp-idf.git "$IDF_PATH"
 fi
 
-# Espressif downloads very slow, fetch from SmingTools
-mkdir -p $IDF_TOOLS_PATH
-ESPTOOLS=esp32-tools-linux-4.1.zip
-$WGET $SMINGTOOLS/$ESPTOOLS -O $DOWNLOADS/$ESPTOOLS
-unzip $DOWNLOADS/$ESPTOOLS -d $IDF_TOOLS_PATH/dist
+# Apply IDF patches
+IDF_PATCH="$(dirname $BASH_SOURCE)/idf.patch"
+pushd "$IDF_PATH"
+git apply -v --ignore-whitespace --whitespace=nowarn "$IDF_PATCH"
 
```

Comment:
@slaff We have a choice of either applying patches or cloning a forked repo. Forking would give more control, but if we go that route it should live in SmingHub.

Reply:
For example, cloning a `sming-4.3` branch means fixes can be applied using `git pull` rather than a re-install.